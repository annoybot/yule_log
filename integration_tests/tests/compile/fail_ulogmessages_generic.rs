use yule_log::{ULogMessages, ULogData};

#[derive(ULogData)]
pub struct A {
    value: u64,
}

#[derive(ULogMessages)]
pub enum LoggedMessages<T> {
    A(A),
}

fn main() {}
