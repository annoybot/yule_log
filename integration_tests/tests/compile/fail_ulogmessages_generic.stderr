error: ULogMessages derive cannot be used with generic enums or where clauses.
 --> tests/compile/fail_ulogmessages_generic.rs:9:10
  |
9 | pub enum LoggedMessages<T> {
  |          ^^^^^^^^^^^^^^

error[E0392]: type parameter `T` is never used
 --> tests/compile/fail_ulogmessages_generic.rs:9:25
  |
9 | pub enum LoggedMessages<T> {
  |                         ^ unused type parameter
  |
  = help: consider removing `T`, referring to it in a field, or using a marker such as `PhantomData`
  = help: if you intended `T` to be a const parameter, use `const T: /* Type */` instead
